packages/ckeditor5-source-editing/src/sourceediting.js

```diff
@@ -241,7 +241,7 @@ export default class SourceEditing extends Plugin {
 		const data = {};
 
 		for ( const [ rootName, domSourceEditingElementWrapper ] of this._replacedRoots ) {
-			const oldData = editor.data.get( { rootName } );
+			const oldData = formatSource( editor.data.get( { rootName } ) );
 			const newData = domSourceEditingElementWrapper.dataset.value;
```

Comment:
We could use the formatter here as well. I'm not sure if the current one would work for the already converted data.

Reply:
Unfortunately, using a formatter twice (for already converted data) will not work here. Moreover, preparing the reverse operation for formatting (de-formatting? 😅) would be non-trivial and error-prone, so I think we should just compare the initial (formatted) data with the current one, whatever it is at the time of exiting from the source editing mode.

Reply:
In this case it would be good to add a proper comment to the formatting function :)